lib/get-config.js

```diff
@@ -303,7 +307,56 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+function _determineConfigForSeverity(config) {
+  switch (config) {
+    case 'off':
+      return { config: false, severity: IGNORE_SEVERITY };
+    case 'warn':
+      return { config: true, severity: WARNING_SEVERITY };
+    case 'error':
+      return { config: true, severity: ERROR_SEVERITY };
+  }
+}
+
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * @param {*} configData
+ */
```

Comment:
Can you add examples for all of the legacy cases as well? I'm thinking something like these:

```js
{
  'some-rule': 'lol',                   // -> { severity: 2, config: 'lol' }
  'other-thing': ['wat', 'is', 'this'], // { severity: 2, config: ['wat', 'is', 'this'] }
  'hmm-thing-here': { zomg: 'lol' },    // -> { severity: 2, config: { zomg: 'lol' } }
}
```